source/common/network/listener_impl.cc

```diff
@@ -28,13 +29,19 @@ void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr*
       final_local_address = original_local_address;
     }
 
-    // A listener that has the use_original_dst flag set to true can still receive connections
-    // that are NOT redirected using iptables. If a connection was not redirected,
-    // the address and port returned by getOriginalDst() match the listener port.
-    // In this case the listener handles the connection directly and does not hand it off.
-    if (listener->socket_.localAddress()->ip()->port() != final_local_address->ip()->port()) {
+    /**
+     * Hands off redirected connections (from iptables) to the listener associated with the
+     * original destination address. If there is no listener associated with the original
+     * destination address, the connection is handled by the listener that receives it.
+     *
+     * Note: A listener that has the use_original_dst flag set to true can still receive
+     * connections that are NOT redirected using iptables. If a connection was not redirected,
+     * the address returned by getOriginalDst() match the listener address. In this case the
+     * listener handles the connection directly and does not hand it off.
+     */
+    if (listener->socket_.localAddress()->asString() != final_local_address->asString()) {
```

Comment:
Agree with htuch's point. We can start from one address with 0.0.0.0 as the wild card. And then extend to a list if necessary.